src/sudo.rs

```diff
@@ -0,0 +1,199 @@
+/// that has been transfered through the channel for a specific denom.
+/// If the period for a RateLimit has ended, the Flow information is reset.
+///
+/// The channel_value is the current value of the denom for the the channel as
+/// calculated by the caller. This should be the total supply of a denom
+pub fn try_transfer(
+    deps: DepsMut,
+    path: &Path,
+    channel_value: Uint256,
+    funds: Uint256,
+    direction: FlowType,
+    now: Timestamp,
+) -> Result<Response, ContractError> {
+    // Sudo call. Only go modules should be allowed to access this
+
+    // Fetch potential trackers for "any" channel of the required token
+    let any_path = Path::new("any", path.denom.clone());
+    let mut any_trackers = RATE_LIMIT_TRACKERS
+        .may_load(deps.storage, any_path.clone().into())?
+        .unwrap_or_default();
+    // Fetch trackers for the requested path
+    let mut trackers = RATE_LIMIT_TRACKERS
+        .may_load(deps.storage, path.into())?
+        .unwrap_or_default();
+
+    let not_configured = trackers.is_empty() && any_trackers.is_empty();
+
+    if not_configured {
+        // No Quota configured for the current path. Allowing all messages.
+        return Ok(Response::new()
+            .add_attribute("method", "try_transfer")
+            .add_attribute("channel_id", path.channel.to_string())
+            .add_attribute("denom", path.denom.to_string())
+            .add_attribute("quota", "none"));
+    }
+
+    // If any of the RateLimits fails, allow_transfer() will return
+    // ContractError::RateLimitExceded, which we'll propagate out
+    let results: Vec<RateLimit> = trackers
+        .iter_mut()
+        .map(|limit| limit.allow_transfer(path, &direction, funds, channel_value, now))
+        .collect::<Result<_, ContractError>>()?;
+
+    let any_results: Vec<RateLimit> = any_trackers
+        .iter_mut()
+        .map(|limit| limit.allow_transfer(path, &direction, funds, channel_value, now))
+        .collect::<Result<_, ContractError>>()?;
+
+    RATE_LIMIT_TRACKERS.save(deps.storage, path.into(), &results)?;
+    RATE_LIMIT_TRACKERS.save(deps.storage, any_path.into(), &any_results)?;
+
+    let response = Response::new()
+        .add_attribute("method", "try_transfer")
+        .add_attribute("channel_id", path.channel.to_string())
+        .add_attribute("denom", path.denom.to_string());
+
+    // Adds the attributes for each path to the response. In prod, the
+    // addtribute add_rate_limit_attributes is a noop
+    let response: Result<Response, ContractError> =
+        any_results.iter().try_fold(response, |acc, result| {
+            Ok(add_rate_limit_attributes(acc, result))
+        });
+
+    results.iter().try_fold(response?, |acc, result| {
+        Ok(add_rate_limit_attributes(acc, result))
+    })
+}
+
+// #[cfg(any(feature = "verbose_responses", test))]
+fn add_rate_limit_attributes(response: Response, result: &RateLimit) -> Response {
```

Comment:
it's not a noop since #[cfg(any(feature = "verbose_responses", test))] is commented out for the add_rate_limit_attributes fn and not applied for calls, right?

Reply:
https://github.com/neutron-org/rate-limiter-contract/blob/b4463a370b25438e83b65f8e26310191b713c737/src/sudo.rs#L139-L149

It's commented out because conditional compilation is broken in cosmwasm optimisors. We can  uncommented it out as soon as it's fixed in optimizors